source/patterns/@aws-solutions-constructs/aws-fargate-kinesisstreams/test/fargate-kinesisstreams.test.ts

```diff
@@ -0,0 +1,612 @@
+/**
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+
+import { FargateToKinesisStreams } from "../lib";
+import * as cdk from "aws-cdk-lib";
+import * as kinesis from 'aws-cdk-lib/aws-kinesis';
+import * as defaults from '@aws-solutions-constructs/core';
+import '@aws-cdk/assert/jest';
+import { Match, Template } from "aws-cdk-lib/assertions";
+
+test('Default construct has all expected properties', () => {
+  const stack = new cdk.Stack();
+
+  const construct = new FargateToKinesisStreams(stack, 'test-fargate-kinesisstreams', {
+    publicApi: false,
+    ecrRepositoryArn: defaults.fakeEcrRepoArn,
+  });
+
+  expect(construct.vpc).toBeDefined();
+  expect(construct.service).toBeDefined();
+  expect(construct.container).toBeDefined();
+  expect(construct.kinesisStream).toBeDefined();
+  expect(construct.cloudwatchAlarms).toBeDefined();
+});
+
+test('Construct deploys Fargate Service in isolated subnets when publicApi is set to false', () => {
+  const stack = new cdk.Stack();
+
+  new FargateToKinesisStreams(stack, 'test-fargate-kinesisstreams', {
+    publicApi: false,
+    ecrRepositoryArn: defaults.fakeEcrRepoArn,
+  });
+
+  const template = Template.fromStack(stack);
+
+  // The default isolated VPC should have two subnets, 2 route tables, and no nat/internet gateway, or routes
+  expect(stack).toCountResources('AWS::EC2::VPC', 1);
+  expect(stack).toCountResources('AWS::EC2::Subnet', 2);
+  expect(stack).toCountResources('AWS::EC2::RouteTable', 2);
+  expect(stack).toCountResources('AWS::EC2::Route', 0);
+  expect(stack).toCountResources('AWS::EC2::NatGateway', 0);
+  expect(stack).toCountResources('AWS::EC2::InternetGateway', 0);
+
+  // Get the first isolated subnet
+  const subnet1Resource = template.findResources('AWS::EC2::Subnet', {
+    Properties: {
+      MapPublicIpOnLaunch: false,
+      Tags: [
+        {
+          Key: Match.exact("Name"),
+          Value: Match.exact("Default/Vpc/IsolatedSubnet1")
```

Comment:
Doest this create a dependency on the default VPC? Because millions of accounts don't have a default VPC (mine, for instance)

Reply:
The Tag name value is a little misleading. This is not the account's default VPC, but instead a new VPC created from the `ec2.Vpc` construct, that has a set of default subnets in it.

Reply:
OK